// components/DaysTimer/Timer.jsx
/* eslint-disable */
import React, { PropTypes } from 'react';
import moment from 'moment';


const propTypes = {
    startDate: PropTypes.string.isRequired
};

class DaysTimer extends React.Component {

    constructor() {
        super();
        // set initial time:y

        this.state={
            days:0,
            hours:0,
            minutes:0,
            seconds:0,
            close:false
        };

    }

    componentDidMount() {
        // update time every second
        this.timer = setInterval(() => {
            this.renderMatchCounter();
        }, 1000);
    }

    componentWillUnmount() {
        window.clearInterval(this.timer);
    }

    renderMatchCounter() {

            let days, hours, minutes, seconds;
            let now = moment();
            //if we have to close 1 hour before startDate
            let startDate = moment(this.props.startDate);

            // let timer = setInterval(function(){
            //if now & startDate are same , means we have to close timer
            if(moment(now).isSame(startDate)){
                window.clearInterval(this.timer);
                this.setState({close:true});

            }
            if (moment(now).isBefore(startDate)) {
                let now = moment();
                let diff = moment.duration(moment(startDate).diff(moment(now)));
                days = parseInt(diff.asDays(),10);
                hours = parseInt(diff.asHours(),10) - days * 24;
                minutes = parseInt(diff.asMinutes(),10) - (days * 24 * 60 + hours * 60);
                seconds = parseInt(diff.asSeconds(),10) - (days * 24 * 60 * 60 + minutes * 60 + hours * 60 * 60);
                this.setState({
                    days:days,
                    hours:hours,
                    minutes:minutes,
                    seconds:seconds
                })
            }



    }

    render() {

        return (
            <div>
               <div className="timer-head">
                    <div className="td-timer">{this.state.days}  - Days</div>
                    <div className="td-timer">{this.state.hours} - hours</div>
                    <div className="td-timer">{this.state.minutes} - Min</div>
                    <div className="td-timer">{this.state.seconds} - Sec</div>
                </div>
            </div>
        );
    }


}

DaysTimer.propTypes = propTypes;

export default DaysTimer;
